compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -620,22 +611,6 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         match self.tcx.hir().get_if_local(impl_def_id) {
                             // Unmet obligation comes from a `derive` macro, point at it once to
                             // avoid multiple span labels pointing at the same place.
-                            Some(Node::Item(hir::Item {
```

Comment:
I'm pretty convinced that this code will never be triggered, but if it is, someone will report it as an ICE :sweat_smile:

Reply:
For this to be triggered, we need to have a nested obligation of a trait be reported as an unsatisfied predicate for a method to be applicable. We only construct nested trait obligations with `ImplDerivedObligation` causes for auto traits, which have no methods.

Reply:
#105732 is here,
maybe we should add an `delay_span_bug` here for match with `Some(node)`?
@compiler-errors